eoscompanion/routes.py

```diff
@@ -161,19 +160,419 @@ def desktop_id_to_app_id(desktop_id):
+    return source['detail'][_SOURCE_IDENTIFIER_KEYS_BY_TYPE[source['type']]]
+
+
+def _serialize_article_content_feed_model(model, app_id, query):
+    '''Serialize the ARTICLE_CARD type content feed model.'''
+    return [{
+        'itemType': 'article',
+        'source': {
+            'type': 'application',
+            'detail': {
+                'applicationId': app_id
+            },
+        },
+        'detail': {
+            'title': model.get_property('title'),
+            'synopsis': model.get_property('synopsis'),
+            'thumbnail': format_thumbnail_uri(
+                app_id,
+                model.get_property('thumbnail-uri'),
+                query['deviceUUID']
+            ),
+            'uri': format_content_data_uri(
+                parse_uri_path_basename(model.get_property('uri')),
+                app_id,
+                query
+            ),
+            'contentType': 'text/html'
+        }
+    }]
+
+
+def maybe_get_property(model, prop_name):
+    '''Get the year from the first-date property of the model, if it exists.'''
+    try:
+        return model.get_property(prop_name)
+    except TypeError:
+        return None
+
+
+def _serialize_artwork_content_feed_model(model, app_id, query):
+    '''Serialize the ARTWORK_CARD type content feed model.'''
+    return [{
+        'itemType': 'artwork',
+        'source': {
+            'type': 'application',
+            'detail': {
+                'applicationId': app_id
+            },
+        },
+        'detail': {
+            'title': model.get_property('title'),
+            'author': model.get_property('author'),
+            'firstDate': maybe_get_property(model, 'first-date'),
+            'thumbnail': format_thumbnail_uri(
+                app_id,
+                model.get_property('thumbnail-uri'),
+                query['deviceUUID']
+            ),
+            'uri': format_content_data_uri(
+                parse_uri_path_basename(model.get_property('uri')),
+                app_id,
+                query
+            ),
+            'contentType': 'text/html'
+        }
+    }]
+
+
+def _serialize_video_content_feed_model(model, app_id, query):
+    '''Serialize the VIDEO_CARD type content feed model.'''
+    return [{
+        'itemType': 'video',
+        'source': {
+            'type': 'application',
+            'detail': {
+                'applicationId': app_id
+            },
+        },
+        'detail': {
+            'title': model.get_property('title'),
+            'thumbnail': format_thumbnail_uri(
+                app_id,
+                model.get_property('thumbnail-uri'),
+                query['deviceUUID']
+            ),
+            'uri': format_content_data_uri(
+                parse_uri_path_basename(model.get_property('uri')),
+                app_id,
+                query
+            ),
+            'duration': model.get_property('duration'),
+            'contentType': 'video/webm'
```

Comment:
Do we not have this as part of the model? It feels fragile to hardcode it here (also for the other types)

Reply:
It does feel fragile.

I'm not sure if it is part of the returned discovery feed models, but it certainly exists in the DmContent models. I'll have to add it to libcontentfeed I guess...